src/commands/syncContent/export/export.ts

```diff
@@ -0,0 +1,128 @@
+            alias: "sd",
+            implies: itemsFilterParams.filter(i => i !== "items"),
+          })
+          .option("language", {
+            type: "string",
+            describe: "Defines language, for which content items will be migrated.",
+            demandOption: "You need to provide language.",
+            alias: "l",
+          })
+          .option("items", {
+            type: "array",
+            string: true,
+            describe: "Array of content items codenames to be migrated.",
+            alias: "i",
+            conflicts: itemsFilterParams.filter(p => p !== "items"),
+          })
+          .option("filename", {
+            type: "string",
+            describe: "Filename where items should be stored.",
+            alias: "f",
+          })
+          .option("last", {
+            type: "number",
+            describe: "Migrate x lastly modified items",
+            conflicts: itemsFilterParams.filter(p => p !== "last"),
+          })
+          .option("byTypesCodenames", {
+            type: "array",
+            string: true,
+            describe: "Migrate items of specified content types",
+            conflicts: itemsFilterParams.filter(p => p !== "byTypesCodenames"),
+          })
+          .option("depth", {
+            type: "number",
+            describe:
+              "Determines the level of linked items in Delivery API response. We encourage using with --limit to prevent hiting upper response limit.",
+            conflicts: ["filter"],
+          })
+          .option("limit", {
+            type: "number",
+            describe:
+              "Updates the limit for number of content items in one Delivery API response. For more information read the Delivery API docs.",
+            conflicts: ["filter"],
+          })
+          .option("filter", {
+            type: "string",
+            describe: "A filter to obtain a subset of items codenames. See Delivery sdk for more info",
+            conflicts: itemsFilterParams.filter(p => p !== "filter"),
+          })
+          .option("skipConfirmation", {
+            type: "boolean",
+            describe: "Skip confirmation message.",
+          }),
+      handler: args => generateSyncContent(args).catch(simplifyErrors),
+    },
+  );
+
+export type GenerateSyncContentParams =
+  & Readonly<{
+    sourceEnvironmentId: string;
+    sourceApiKey: string;
+    sourceDeliveryKey?: string;
+    language: string;
+    filename?: string;
+    items?: ReadonlyArray<string>;
+    last?: number;
+    depth?: number;
+    limit?: number;
+    byTypeCodenames?: ReadonlyArray<string>;
+    filter?: string;
+    skipConfirmation?: boolean;
+  }>
+  & LogOptions;
+
+export const generateSyncContent = async (params: GenerateSyncContentParams) => {
+  const deliveryClient = createDeliveryClient({
+    environmentId: params.sourceEnvironmentId,
+    previewApiKey: params.sourceDeliveryKey,
```

Comment:
In retrospect, it doesn't make sense to get the items from the published delivery endpoint for syncing content. You most likely also want the not-yet-published content as well. So I suggest we get rid of the `sourceDeliverySecureKey` parameter and always enable the preview mode in the delivery client. The `sourceDeliveryPreviewKey` will remain optional, because not all parameters make a delivery API call, but it will be necessary for all delivery API calls. I am sorry for the initial confusion.

Reply:
You left the `sourceDeliverySecureKey` in the `GenerateSyncContentParams` type.